### PR TITLE
[FR] Optionally pass nested schema for further validation

### PR DESCRIPTION
**Pull request by Mikaayenson**
## Issues
N/A

## Details
*Describe details about your requested changes*

- We want to validate functions that use nested fields. This WIP enables us to pass down the schema for os-specific nested fields and validate nested fields belong to a parent field (and that the nested fields existed per stack).
- May want to remove py27 from the ci as well.
Screenshot 2023-06-24 at 7 36 26 AM

### CODE REVIEW

**Comment by Mikaayenson** [comment]
> Looks like it's `ubuntu-latest` that's getting in the way of your Python 2.7 validation

Yup. May have to let it go based on the [deprecation](https://github.blog/changelog/2022-08-09-github-actions-ubuntu-22-04-is-now-generally-available-on-github-hosted-runners/).

File: .github/workflows/pythonpackage.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ on:
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
```
Comment (Mikaayenson) [inline .github/workflows/pythonpackage.yml:15]:
```suggestion
    runs-on: ubuntu-22.04
```

**Review by eric-forte-elastic — APPROVED** [review]

**Review by terrancedejesus — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,14 @@
 # Event Query Language - Changelog
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+# Version 0.9.16
+
+_Released 2023-07-12_
+
+### Added
+
+* Support for schema validation on nested fields
```
Comment (brokensound77) [inline CHANGELOG.md:10]:
```suggestion
* Support for exposing nested fields for validation
```

File: eql/signatures.py [hunk 3]
```diff
@@ -10,7 +10,7 @@ class SignatureMixin(object):
     additional_types = None
 
     @classmethod
-    def validate(cls, arguments):  # type: (list[NodeInfo]) -> int
+    def validate(cls, arguments, event_type=None, schema=None):  # type: (list[NodeInfo]) -> int
```
Comment (brokensound77) [inline eql/signatures.py:13]: update type hinting

```suggestion
    def validate(cls, arguments, event_type=None, schema=None):  # type: (list[NodeInfo, str, dict]) -> int
```

**Review by brokensound77 — CHANGES_REQUESTED** [review]
hold off merging until further discussions

**Comment by Mikaayenson** [comment]
Hijacking https://github.com/endgameinc/eql/pull/71

**Pull request closed without merging**